rust/headless-client/src/ipc_service.rs

```diff
@@ -505,11 +528,18 @@ impl<'a> Handler<'a> {
 
                 session.connlib.set_disabled_resources(disabled_resources);
             }
-            ClientMsg::StartTelemetry { environment } => self.telemetry.start(
-                &environment,
-                firezone_bin_shared::git_version!("gui-client-*"),
-                firezone_telemetry::IPC_SERVICE_DSN,
-            ),
+            ClientMsg::StartTelemetry { environment } => {
+                let daemon_version = git_version().to_string();
+                self.telemetry.start(
+                    &environment,
+                    git_version(),
+                    firezone_telemetry::IPC_SERVICE_DSN,
+                );
+                self.ipc_tx
+                    .send(&ServerMsg::Handshake { daemon_version })
```

Comment:
Why is this tied to starting telemetry? Shouldn't we just send this as soon as we connect? Something like a `Hello` message? Once we make telemetry opt-out, we'll have to change this to still send it which is kind of odd?

Reply:
Perhaps these things can be merged into like `ClientHello`, `ServerHello`, inspired by TLS.

`ClientHello` could contain:

- Enable telemetry yes / no
- Which version client is running
- ... 

`ServerHello` "answers" with:

- Which version it is running
- ...

Reply:
Yeah changing the socket name sounds better. It'll conflate an "IPC service isn't running" error with a "IPC service needs restart" error but then there won't be any risk of protocol mismatch.

The reason I had this design is so we'd get a non-fatal error if there's a mismatch and the service is behind. If the GUI initiated this exchange, then the service wouldn't recognize the new message and it would drop the IPC connection